### ISSUE TITLE
context manager scope + suppress = code unreachable in pyright >= 1.1.100

### ISSUE THREAD

**Issue by askurihin** [microsoft/pyright#1376]
**Describe the bug**
Pyright reports code is unreachable after suppress scope ends.
This was introduced in pyright 1.1.100 (pyright 1.1.99 does not have this problem)

**To Reproduce**
```python
from contextlib import suppress

some_dict = dict()
some_string = "HELLO"

with suppress(KeyError):
    print(some_dict['missing_key'])

print(some_string.lower())

```

```sh
❯ pyright test.py     
  9:19 - error: Cannot access member "lower" for type "Never"
    Type "Never" is unsupported (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.593sec
```

**Expected behavior**
pyright should correctly understand where context manager scope ends.

**Version**
command-line pyright @ 1.1.101

**Comment by erictraut** [comment]
Thanks for the bug report. You are correct, I introduced a regression in the latest release when fixing another bug. The nature of the regression was such that it was not caught by our unit tests. I've fixed the problem and improved our unit tests so similar problems will be caught in the future. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.102, which I just published. It will also be included in the next release of pylance.
